komodo/prettier.py

```diff
@@ -0,0 +1,149 @@
+    try:
+        basestring
+    except NameError:
+        basestring = str # No basestring on Python3
+
+    if all([isinstance(package, basestring) for package in config.values()]):
+        return False
+    elif all([isinstance(package, ruamel.yaml.comments.CommentedMap)
+              for package in config.values()]):
+        return True
+
+    raise ValueError(
+        'Inconsistent configuration file. '
+        'Not able to detect if it is a release or repository.'
+    )
+
+
+def prettier(yaml_input_string):
+    '''Takes in a string corresponding to a YAML Komodo configuration, and returns
+    the corresponding prettified YAML string.'''
+
+    ruamel_instance = ruamel.yaml.YAML()
+    ruamel_instance.indent(  # Komodo prefers two space indendation
+        mapping=2, sequence=4, offset=2
+    )
+    ruamel_instance.width = 1000  # Avoid ruamel wrapping long
+
+    config = ruamel_instance.load(yaml_input_string)
+
+    komodo_repository = is_repository(config)
+
+    # On Python3.6+, sorted_config can just be an
+    # ordinary dict as insertion order is then preserved.
+    sorted_config = ruamel.yaml.comments.CommentedMap()
+    for package in sorted(config, key=str.lower):
+        sorted_config[package] = config[package]
+
+    setattr(sorted_config, ruamel.yaml.comments.comment_attrib, config.ca)
+
+    yaml_output = StringIO()
+    ruamel_instance.dump(
+        sorted_config,
+        yaml_output,
+        transform=functools.partial(repository_specific_formatting, komodo_repository)
+    )
+
+    if sys.version_info < (3, 0):
+        # Need to encode the byte-string on Python2
+        return yaml_output.getvalue().encode('utf-8')
+
+    return yaml_output.getvalue()
+
+
+def prettified_yaml(filepath, check_only=True):
+    '''Returns `True` if the file is already "prettified", `False` otherwise.
+    If `check_only` is False, the input file will be "prettified" in place if necessary.
+    '''
+
+    print('Checking {}... '.format(filepath), end='')
+
+    with open(filepath, "r") as fh:
+        yaml_input_string = fh.read()
+
+    yaml_prettified_string = prettier(yaml_input_string)
+
+    if yaml_prettified_string != yaml_input_string:
+        print('{} reformatted!'.format('would be' if check_only else ''))
+        if not check_only:
+            with open(filepath, 'w') as fh:
+                fh.write(yaml_prettified_string)
+        return False
+
+    print('looking good!')
+    return True
+
+
+def prettier_main():
+    '''Main function doing user argument parsing and calling necessary functions.
+    '''
+
+    parser = argparse.ArgumentParser(
+        description=(
+            'Check and/or format the Komodo configuration files. '
+            'Takes in any number of yml files, which could be e.g. the main '
+            'Komodo repository and an arbitrary number of releases. '
+            'Throws a hard error if the same package is defined multiple times.'
+        )
+    )
+    parser.add_argument(
```

Comment:
to avoid hard exit and weired stacktraces when inputting a folder, you could do something like this
```
def is_valid_file(parser, arg):
    if not os.path.isfile(arg):
        parser.error("{} is not a file!".format(arg))
    else:
        return arg

parser.add_argument(
    "files",
    type=lambda path: is_valid_file(parser, path),
    nargs="+",
    help="One or more files to format/check",
)
```